### PR TITLE
[layout] Automatic conversion of 2.x composition templates to layouts

### PR DESCRIPTION
**Pull request by elpaso**

### CODE REVIEW

File: src/core/layout/qgscompositionconverter.h [hunk 1]
```diff
@@ -144,6 +144,23 @@ class CORE_EXPORT QgsCompositionConverter
         QPointF *position = nullptr,
         bool pasteInPlace = false );
 
+    /**
+     * Check if the given \a document is a composition template
+     * \param document a dom document
```
Comment (nyalldawson) [inline src/core/layout/qgscompositionconverter.h]: Drop this param annotation - it's not providing any new info

File: src/core/layout/qgscompositionconverter.h [hunk 2]
```diff
@@ -144,6 +144,23 @@ class CORE_EXPORT QgsCompositionConverter
...
+     * \return true if the document is a composition template
+     */
```
Comment (nyalldawson) [inline src/core/layout/qgscompositionconverter.h]: \since QGIS 3.0.1

**Review by nyalldawson — CHANGES_REQUESTED** [review]
Looking great! 

Please check expected_importComposerTemplateMap_0_mask.png -- I think it's been updated with the wrong reference image and is now passing any image.

**Comment by elpaso** [comment]
Yes, I noticed and triple checked the wierd mask yesterday evening but I couldn't figure out where the problem was.  I'll give it another fresh look.

**Comment by nyalldawson** [comment]
Looks good! +1 to merge if Travis approves

**Pull request merged**